### PR TITLE
feat: sitemap detection audit

### PR DESCRIPTION
**Pull request by AndreiAlexandruParaschiv**
Sitemap detection for EDS customers
Ticket: https://jira.corp.adobe.com/browse/SITES-18817

Related PRs:

- https://github.com/adobe/spacecat-api-service/pull/136
- https://github.com/adobe/spacecat-audit-post-processor/pull/67

### CODE REVIEW

File: src/sitemap/handler.js [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+      success: true,
+      reasons: logMessages,
+    };
+  }
+
+  console.log(logMessages.join(' '));
+  return {
+    success: false,
+    reasons: logMessages,
+  };
```
[the remark this answers is not in the document]
Comment (AndreiAlexandruParaschiv) [reply src/sitemap/handler.js:185]: Make sense, I've added the first sitemap.xml path that I could find. 
We could tackle going deeper into the sitemap_index in the dedicated story if it makes sense. The scenario is: the sitemap exists but it's an index of sitemaps - we should include all sitemaps xmls (maybe as an object to differentiate between them)

File: src/sitemap/handler.js [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+async function checkSitemap(sitemapUrl) {
+  try {
+    const sitemapContent = await fetchContent(sitemapUrl);
+    if (!sitemapContent) {
+      return {
+        existsAndIsValid: false,
+        reasons: [ERROR_CODES.SITEMAP_NOT_FOUND, ERROR_CODES.SITEMAP_EMPTY],
+      };
+    }
+    const isValidXml = sitemapContent.trim().startsWith('<?xml');
```
[the remark this answers is not in the document]
Comment (AndreiAlexandruParaschiv) [reply src/sitemap/handler.js:79]: That's a good idea, but I think it should be tackled in a separate PR as this one already has additional checks and optimizations beyond the scope of the initial ticket.

File: src/sitemap/handler.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+      log.error(`No site with siteId "${siteId}" exists.`);
+      return notFound('Site not found');
+    }
+
+    const baseURL = site.getBaseURL();
+    const auditResult = await findSitemap(baseURL);
+
+    log.info(`Audit result for ${baseURL}:\n${JSON.stringify(auditResult, null, 2)}`);
+
+    await sqs.sendMessage(queueUrl, {
```
Comment (ekremney) [inline src/sitemap/handler.js:122]: just a heads-up: we started saving the audit result before sending the sqs message. Example here: https://github.com/adobe/spacecat-audit-worker/blob/main/src/experimentation/handler.js#L77

**Review by ekremney — APPROVED** [review]

**Pull request merged**